data_analysis/static_gtfs_analysis.py

```diff
@@ -333,23 +344,99 @@ def download_zip(version_id: str) -> zipfile.ZipFile:
     return CTA_GTFS
 
 
-def main(version_id: str = None) -> geopandas.GeoDataFrame:
-    """Download data from CTA, construct shapes from shape data,
-    and save to geojson file
+def download_extract_format(version_id: str) -> GTFSFeed:
+    """Download a zipfile of GTFS data for a given version_id,
+        extract data, and format date column.
 
     Args:
-        version_id (str, optional): The schedule version to use.
-            Defaults to None.
+        version_id (str): The version of the GTFS schedule data to download
 
     Returns:
-        geopandas.GeoDataFrame: DataFrame with route shapes
+        GTFSFeed: A GTFSFeed object with formated dates
     """
-    if version_id is None:
-        version_id = VERSION_ID
-
     CTA_GTFS = download_zip(version_id)
     data = GTFSFeed.extract_data(CTA_GTFS, version_id=version_id)
     data = format_dates_hours(data)
+    return data
+
+
+def create_GTFSFeeds_list(schedule_list: List[dict]) -> List[GTFSFeed]:
+    """Make a list of GTFS feeds from a list of schedule versions
+
+    Args:
+        schedule_list (List[dict]): A list of schedule versions and their
+            start and end dates
+
+    Returns:
+        List[GTFSFeed]: A list of GTFSFeed objects for the
+            schedule versions in schedule_list
+    """
+    version_list = [value for d in schedule_list for key, value in d.items()
+                    if key == 'schedule_version']
+    CTA_GTFS_list = []
+    for version in version_list:
+        CTA_GTFS_list.append(download_extract_format(version))
+    return CTA_GTFS_list
+
+
+def concat_GTFSFeeds(schedule_list: List[dict]) -> GTFSFeed:
```

Comment:
Yeah I will probably just delete this. At first, I thought each schedule version was needed, but I really only need one version so that the `final_gdf` `DataFrame` that has the GPS coordinates can be generated. This is what is used for creating the map colored by the trip ratios. I left a comment on that [here](https://github.com/chihacknight/chn-ghost-buses/blob/e707c3b56c203a9e38c56a9a32e7a4948ef2ee25/data_analysis/static_gtfs_analysis.py#L428-L433) because I was unsure about it.